lineax/_solver/Woodbury.py

```diff
@@ -0,0 +1,138 @@
+from typing import Any
+from typing_extensions import TypeAlias
+
+import jax
+import jax.numpy as jnp
+from jaxtyping import Array, PyTree
+
+from .._operator import (
+    AbstractLinearOperator,
+    is_Woodbury,
+    MatrixLinearOperator,
+    woodbury,
+)
+from .._solution import RESULTS
+from .._solve import AbstractLinearSolver, AutoLinearSolver
+from .misc import (
+    pack_structures,
+    PackedStructures,
+    ravel_vector,
+    transpose_packed_structures,
+    unravel_solution,
+)
+
+
+_Woodbury_State: TypeAlias = tuple[
+    tuple[Array, Array, Array],
+    tuple[AbstractLinearSolver, Any, AbstractLinearSolver, Any],
+    PackedStructures,
+]
+
+
+def compute_pushthrough(A_solver, A_state, C, U, V):
+    # Push through ( C^-1 + V A^-1 U) y = x
+    vmapped_solve = jax.vmap(
+        lambda x_vec: A_solver.compute(A_state, x_vec, {})[0], in_axes=1, out_axes=1
+    )
+    pushthrough_mat = jnp.linalg.inv(C) + V @ vmapped_solve(U)
```

Comment:
Should we make `C` an `AbstractLinearOperator` as well?

Reply:
I originally had C as an AbstractLinearOperator but it always had to be as_matrix() so its inverse can be computed by jnp.linalg. Therefore, it made the most sense that it is given as a matrix at input so that as_matrix operation is not hidden. If some linearoperators could have an inverse method then perhaps it would make sense for C to be an operator?

Reply:
Ah, right!

Yes, you're completely correct. Indeed I think the appropriate thing to do would be to have a built-in notion of the inverse of an operator. (Tagging also @f0uriest as we were discussing this in #96.)

The inverse of an operator is another operator, and it is basically defined by the action of a linear solve. I think that should mean:
```python
class InverseLinearOperator(AbstractLinearOperator):
    operator: AbstractLinearOperator
    solver: AbstractLinearSolver = AutoLinearSolver(well_posed=True)

    def mv(self, vector):
        return linear_solve(self.operator, vector)

   ...  # other methods here
```

Reply:
Ok, this seems fair. I tried to implement this way and just wanted to raise a two points which you might have comments on.

1. The InverseLinearOperator has to go in _solve.py (or at least not _operator.py) to avoid circular imports. This is potentially a little confusing but maybe it is really the right place for it.
2. Having no conjugate method for LinearOperators makes for a bit of head-ache for this one. conj is attached to the solver so you can update the state of the solver for a conjugate inside InverseLinearOperator (similarly for transpose). However, when you do it this way your operator is not updated. The most obvious way to me to implement the as_matrix method was to use jnp.linalg.inv(self.operator.as_matrix()) - but this ignores the solver_state which might have been transposed or conjugated. I suppose my question is, should LinearOperators have a conjugate method or should LinearSolvers have an as_matrix method for the inverse?

This probably should be its own issue, PR, etc.